lib/viewcomponents/gui/toolbar.jsx

```diff
@@ -0,0 +1,94 @@
+var React = require('react');
+var classSet = React.addons.classSet;
+
+var ObjectUtils = require('../../utilities/objects');
+
+var Panel = require('./panel.jsx');
+/*
+tools: an object of tool config objects keyed by name {
+  panelContent: constructed component of contents of the panel,
+  icon: path to icon image
+  title: string title of panel
+  props: any props to pass to the panel
+}
+ */
+
+
+var Toolbar = React.createClass({
+  propTypes: {
+    tools: React.PropTypes.object
+  },
+
+  getDefaultProps: function() {
+    return {
+      tools: {}
+    };
+  },
+
+  getInitialState: function() {
+    return {
+      activePanel: null
+    };
+  },
+
+  render: function() {
+    var createButtons = function() {
+      var buttons = [];
+      ObjectUtils.each(this.props.tools, function(toolConfig, name) {
+
+        var clickHander = function() {
+          // make active panel clicked panel. If already active, close panel (make null).
+          this.setState({
+            activePanel: this.state.activePanel === name ? null : name
+          });
+        }.bind(this);
+
+        var classPredicates = {};
+        classPredicates.button = true;
+        classPredicates.active = this.state.activePanel === name;
+        classPredicates['button-' + name] = !!name;
+        var classes = classSet(classPredicates);
+
+        buttons.push(
```

Comment:
Rather than pushing these to a list, you could use map to get the list. That seems like a nicer way to do it.

your way (paraphrased without ObjectUtils):
```
var newList = [];
oldList.each(function(element) {
  // something something
  newList.push(something);
};
```

what I'm suggesting:
```
var newList = oldList.map(function(element) {
  // something something
  return something;
};
```